From c703b9bc404500b26ffce7f6b6bd2bab291de06c Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario_limonciello@dell.com>
Date: Wed, 15 Jul 2015 16:21:13 -0500
Subject: [PATCH 2/2] Don't call efibootmgr after fwupdate.

fwupdate will handle creating the appropriate BootNext entry.
No other changes are needed from fwupd.
---
 contrib/fwupd.spec.in  |  1 -
 src/fu-provider-uefi.c | 24 ------------------------
 2 files changed, 25 deletions(-)

diff --git a/contrib/fwupd.spec.in b/contrib/fwupd.spec.in
index decd367..e48f055 100644
--- a/contrib/fwupd.spec.in
+++ b/contrib/fwupd.spec.in
@@ -32,7 +32,6 @@ BuildRequires: libappstream-glib-devel
 
 # we'll fix this when Peter has the latest code in a Fedora package
 # BuildRequires: fwupdate-devel >= 0.1
-# Requires: efibootmgr
 
 %description
 fwupd is a daemon to allow session software to update device firmware.
diff --git a/src/fu-provider-uefi.c b/src/fu-provider-uefi.c
index a7d6b39..4ba8d1d 100644
--- a/src/fu-provider-uefi.c
+++ b/src/fu-provider-uefi.c
@@ -238,30 +238,6 @@ fu_provider_uefi_update (FuProvider *provider,
 			     fwup_strerror (fwup_error));
 		goto out;
 	}
-
-	/* schedule our next boot to be the fwupdate */
-	if (!g_spawn_command_line_sync ("/usr/sbin/efibootmgr -n 1337",
-					NULL,
-					&standard_error,
-					&rc,
-					&error_local)) {
-		ret = FALSE;
-		g_set_error (error,
-			     FWUPD_ERROR,
-			     FWUPD_ERROR_NOT_SUPPORTED,
-			     "Failed to launch efibootmgr: %s",
-			     error_local->message);
-		goto out;
-	}
-	if (!g_spawn_check_exit_status (rc, &error_local)) {
-		ret = FALSE;
-		g_set_error (error,
-			     FWUPD_ERROR,
-			     FWUPD_ERROR_NOT_SUPPORTED,
-			     "UEFI firmware update failed: %s",
-			     error_local->message);
-		goto out;
-	}
 out:
 	fwup_resource_iter_destroy (&iter);
 	return ret;
-- 
1.9.1

